common/test/acceptance/tests/lms/test_problem_types.py

```diff
@@ -0,0 +1,752 @@
+        """
+        msg = "Wait for status to be {}".format(status)
+        selector = ', '.join(self.status_indicators[status])
+        self.problem_page.wait_for_element_visibility(selector, msg)
+
+    @abstractmethod
+    def answer_problem(self, correct):
+        """
+        Args:
+            `correct` (bool): Inputs correct answer if True, else inputs
+                incorrect answer.
+        """
+        raise NotImplementedError()
+
+
+class ProblemTypeTestMixin(object):
+    """
+    Test cases shared amongst problem types.
+    """
+    @attr('shard_2')
+    def test_answer_correctly(self):
+        """
+        Scenario: I can answer a problem correctly
+        Given External graders respond "correct"
+        And I am viewing a "<ProblemType>" problem
+        When I answer a "<ProblemType>" problem "correctly"
+        Then my "<ProblemType>" answer is marked "correct"
+        And The "<ProblemType>" problem displays a "correct" answer
+        And a "problem_check" server event is emitted
+        And a "problem_check" browser event is emitted
+        """
+        # Make sure we're looking at the right problem
+        self.assertEqual(self.problem_page.problem_name, self.problem_name)
+
+        # Answer the problem correctly
+        self.answer_problem(True)
+        self.problem_page.click_check()
+        self.wait_for_status('correct')
+
+        # Check for corresponding tracking event
+        expected_events = [
+            {'event_source': 'server', 'event_type': 'problem_check'},
+            {'event_source': 'browser', 'event_type': 'problem_check'},
+        ]
+
+        for event in expected_events:
+            self.wait_for_events(event_filter=event, number_of_matches=1)
+
+    @attr('shard_2')
+    def test_answer_incorrectly(self):
+        """
+        Scenario: I can answer a problem incorrectly
+        Given External graders respond "incorrect"
+        And I am viewing a "<ProblemType>" problem
+        When I answer a "<ProblemType>" problem "incorrectly"
+        Then my "<ProblemType>" answer is marked "incorrect"
+        And The "<ProblemType>" problem displays a "incorrect" answer
+        """
+        # Make sure we're looking at the right problem
+        self.assertEqual(self.problem_page.problem_name, self.problem_name)
+
+        # Answer the problem incorrectly
+        self.answer_problem(False)
+        self.problem_page.click_check()
+        self.wait_for_status('incorrect')
+
+    @attr('shard_2')
+    def test_submit_blank_answer(self):
+        """
+        Scenario: I can submit a blank answer
+        Given I am viewing a "<ProblemType>" problem
+        When I check a problem
+        Then my "<ProblemType>" answer is marked "incorrect"
+        And The "<ProblemType>" problem displays a "blank" answer
+        """
+        # Make sure we're looking at the right problem
+        self.assertEqual(self.problem_page.problem_name, self.problem_name)
```

Comment:
In the setUp for the test,  it installs the problem fixture, then goes to auto_auth_page, the to the courseware_page.  After all that, the problem_page object is instantiated.  The problem_page is sort of a sub-page of the courseware_page.

The purpose of this line is to make sure that the courseware_page actually has the problem on it.   I will add a wait_for around the assert just in case there are any timing issues.